source/velocity_boundary_conditions/gplates.cc

```diff
@@ -220,207 +198,102 @@ namespace aspect
-                                                     / cmyr_si;
+            velocity_values[0][idx_theta][idx_phi]= spherical_velocities[0] * cmyr_si;
+            velocity_values[1][idx_theta][idx_phi]= spherical_velocities[1] * cmyr_si;
 
             i++;
           }
 
+
+        // number of intervals in the direction of theta and phi
+        std_cxx11::array<unsigned int,2> table_intervals;
+        table_intervals[0] = n_theta - 1;
+        table_intervals[1] = n_phi - 1;
+
+
+        // min and max extent of the grid in the direction of theta and phi (whole spheres in GPlates)
+        // polar angle theta: from 0° to 180°(PI)
+        grid_extent[0].first = 0;
+        grid_extent[0].second = numbers::PI;
+        // azimuthal angle phi: from 0° to 360°(2*PI)
+        grid_extent[1].first = 0;
+        grid_extent[1].second = 2 * numbers::PI;
+
+
+        for (unsigned int i = 0; i < 2; i++)
+          {
+            if (velocities[i])
+              delete velocities[i];
+            velocities[i] = new Functions::InterpolatedUniformGridData<2> (grid_extent,
+                                                                           table_intervals,
+                                                                           velocity_values[i]);
+          }
+
         AssertThrow(i == n_points,
                     ExcMessage (std::string("Number of read in points does not match number of points in file. File corrupted?")));
       }
 
       template <int dim>
       Tensor<1,dim>
-      GPlatesLookup::surface_velocity(const Point<dim> &position,
-                                      const double time_weight) const
+      GPlatesLookup<dim>::surface_velocity(const Point<dim> &position) const
       {
-        Tensor<1,3> internal_position;
+        Point<3> internal_position;
         if (dim == 2)
           internal_position = rotation_matrix * convert_tensor<dim,3>(position);
         else
           internal_position = convert_tensor<dim,3>(position);
 
-        // Main work, interpolate velocity at this point
-        const Tensor<1,3> interpolated_velocity = interpolate(internal_position, time_weight);
-
-        Tensor<1,dim> output_boundary_velocity;
+        //transform internal_position in spherical coordinates
+        const std_cxx11::array<double,3> internal_position_in_spher_array =
+          ::aspect::Utilities::spherical_coordinates(internal_position);
 
-        if (dim == 2)
-          // convert_tensor conveniently also handles the projection to the 2D plane by
-          // omitting the z-component of velocity (since the 2D model lies in the x-y plane).
-          output_boundary_velocity = convert_tensor<3,dim>(transpose(rotation_matrix) * interpolated_velocity);
-        else
-          output_boundary_velocity = convert_tensor<3,dim>(interpolated_velocity);
+        //remove the radius (first entry of internal_position_in_spher_array)
+        Point<2> internal_position_in_spher_rad;
```

Comment:
Coud you extend the comment by stating that we also sort the components of the spherical position differently? I think it is from [r,phi,theta] to [theta,phi], but better check.